docs/proposals/kubemark.md

```diff
@@ -0,0 +1,150 @@
+independent, i.e. number of existing Nodes do not impact a performance of a single Node. This is not entirely true, as number of Nodes can increase latency of various
+components on Master machine, which in turn may increase latency of Node operations, but we’re not interested in measuring this effect here. In turn we want to measure how
+number of Nodes and the load imposed by Node daemons affects the performance of Master components.
+
+## Kubemark architecture overview
+
+The high-level idea behind Kubemark is to write library that allows running artificial "Hollow" Nodes that will be able to simulate a behavior of real Kubelet and KubeProxy in a single, lightweight
+binary. Hollow components will need to correctly respond to Controllers (via API server), and preferably, in the fullness of time, be able to ‘replay’
+previously recorded real traffic (this is out of scope for initial version). In the initial version we only want them to be able to fool Master
+components and put some configurable (in what way TBD) load on them. To teach Hollow components replaying recorded traffic they will need
+to store data specifying when given Pod/Container should die (e.g. observed lifetime). Such data can be extracted e.g. from etcd Raft logs,
+or it can be reconstructed from Events.
+
+When we have Hollow Node ready, we’ll be able to test performance of Master Components by creating a real Master Node, with API server, Controllers, etcd and whatnot, and
+create number of Hollow Nodes that will register to the running Master.
+
+For the initial version we will teach Master components to use port number to identify Kubelet/KubeProxy. This will allow running those components on non-default ports, and in
+the same time will allow to run multiple Hollow Nodes on a single machine. During setup we will generate credentials for cluster communication and pass them to
+HollowKubelet/HollowProxy to use. Master will treat all HollowNodes as normal ones.
+
+To make Kubemark easier to maintain when system evolves Hollow components will reuse real "production" code for Kubelet and KubeProxy, but will mock all the backends with no-op or very simple mocks. We belive that this approach is better in the long run than writing special "performance-test-aimed" separate version of them. This may take more time to create an initial version, but we think maintenance cost will be noticeably smaller.
+
+![Kubmark architecture diagram](Kubemark_architecture.png?raw=true "Kubemark architecture overview")
+
+## Things to simulate
+
+In real Kubernetes on a single Node we run two daemons that communicate with Master in some way: Kubelet and KubeProxy. In this paragraph we list what kind of communication we
+have between those daemons and Master components. First step in creating holograms is understanding the communication between Master and Node components.
```

Comment:
Since we decided that we would proceed with using real Kubelet/KubeProxy code with some interefaces mocked, I think this section doesn't make much sense for Kubelet and KubeProxy.
Personally, would leave only Heapster and Elasticsearch sections.
WDYT?
cc @davidopp